linux_pipeline/Jenkinsfile_kernel_bisect_commit_validator

```diff
@@ -1,193 +1,267 @@
-                }
-                success {
-                    nunit(testResultsPattern: 'scripts/azure_kernel_validation/tests.xml')
-                }
+        }
+
+        println "Cleaning Azure resources up..."
+        sh(script: """#!/bin/bash
+                      pushd ./scripts/azure_kernel_validation
+                      bash remove_azure_vm_resources.sh "${BUILD_NAME}${BUILD_NUMBER}${BRANCH_NAME}"
+                      popd
+                      """
+        )
+
+        archiveArtifacts(artifacts: "$BUILD_NAME$BUILD_NUMBER$BRANCH_NAME-boot-diagnostics/*.log", allowEmptyArchive: true)
+        if (exitStatus == 0) {
+            nunit(testResultsPattern: 'scripts/azure_kernel_validation/tests.xml', failIfNoResults: false)
+        } else {
+            nunit(testResultsPattern: 'scripts/azure_kernel_validation/tests-fail.xml', failIfNoResults: false)
+            sh(script: "exit 1")
+        }
+    }
+}
+
+def perf() {
+    PERF_OPTIONS = getPerfOptions(PERF_TEST)
+    XML = perfXml[PERF_TEST]
+    stage('validation_functional_hyperv') {
+        checkout scm
+        withCredentials(bindings: [string(credentialsId: 'LISA_IMAGES_SHARE_URL', variable: 'LISA_IMAGES_SHARE_URL'),
+                                   string(credentialsId: 'LISA_TEST_DEPENDENCIES', variable: 'LISA_TEST_DEPENDENCIES')]) {
+            println 'Running LISA...'
+            unstash 'kernel_artifacts'
+            try {
+                PowerShellWrapper("""
+                    & .\\scripts\\lis_hyperv_platform\\main_perf.ps1
+                        -JobId $BUILD_NAME$BUILD_NUMBER$BRANCH_NAME
+                        -InstanceName $BUILD_NAME$BUILD_NUMBER$BRANCH_NAME
+                        -VHDType $DISTRO_VERSION -WorkingDirectory C:\\workspace
+                        -OSVersion 7.4
+                        -LISAImagesShareUrl $LISA_IMAGES_SHARE_URL
+                        -XmlTest $XML
+                        -LisaTestDependencies $LISA_TEST_DEPENDENCIES
+                        -LocalKernelFolder "scripts/package_building_bisect/$BUILD_NUMBER-$BRANCH_NAME/**/rpm"
+                        -LisaPerfOptions '${PERF_OPTIONS}'
+                    """)
+                println 'Finished running LISA.'
+            } catch (Exception e){
+                println "LISA failed"
+                currentBuild.result = 'FAILURE'
+            } finally {
+                println "$BUILD_NAME$BUILD_NUMBER$BRANCH_NAME\\lis-test\\WS2012R2\\lisa\\TestResults\\**\\*"
+                archiveArtifacts "$BUILD_NAME$BUILD_NUMBER$BRANCH_NAME\\lis-test\\WS2012R2\\lisa\\TestResults\\**\\*"
+                stash includes: "$BUILD_NAME$BUILD_NUMBER$BRANCH_NAME\\lis-test\\WS2012R2\\lisa\\TestResults\\**\\*", name: 'perf_result'
+                junit "$BUILD_NAME$BUILD_NUMBER$BRANCH_NAME\\lis-test\\WS2012R2\\lisa\\TestResults\\**\\*.xml"
             }
         }
     }
 }
+
+def extract_result() {
```

Comment:
give a better name to this stage, extract result does not mean anything:
parse_perf_data maybe? extrac_perf_data? extract_lisa_result?